source/common/upstream/cluster_manager_impl.cc

```diff
@@ -296,6 +296,17 @@ void ClusterManagerImpl::onClusterInit(Cluster& cluster) {
     cluster_data->second->thread_aware_lb_->initialize();
   }
 
+  // If the cluster doesn't manage updates itself, we add a callback here that will trigger the
```

Comment:
I have two high level questions before I review this in detail:
1) I don't recall the history of why the cluster manager manages priority set updates but the cluster manages membership updates. Can you refresh my memory here.
2) If we do want to move managing membership updates to here, why not just do this universally and avoid having the boolean entirely? It seems relatively low risk to do that and will make the code simpler. Then we can add batching logic here if we want?